### PR TITLE
Adding an end point /fqm/migrate

### PR DESCRIPTION
**Pull request by kjain110**
Screen Shot 2024-09-09 at 4 21 13 PM

### CODE REVIEW

File: src/main/java/org/folio/fqm/resource/MigrationController.java [hunk 1]
```diff
@@ -15,4 +19,24 @@ public class MigrationController implements FqmVersionApi {
...
+      fqmMigrateRequest.getFields()
+    );
+
+    migrationService.migrate(migratableQueryInformation);
+
+    FqmMigrateResponse fqmMigrateResponse = new FqmMigrateResponse()
+      .entityTypeId(migratableQueryInformation.entityTypeId())
+      .fqlQuery(migratableQueryInformation.fqlQuery())
+      .fields(migratableQueryInformation.fields())
+      .warnings(migratableQueryInformation.warnings().stream()
```
Comment (ncovercash) [inline src/main/java/org/folio/fqm/resource/MigrationController.java:37]: I'd like to return the type and description in the API response, so the description can be returned to the user and the type can be used by the receiving module/etc to handle the case appropriately

File: src/main/resources/swagger.api/schemas/FqmMigrateResponse.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        "type": "string"
+      }
+    },
+  "warnings": {
+    "description": "List of warnings issued during the upgrade",
+    "type": "array",
+    "items": {
+      "type": "string"
+    }
+  }
```
Comment (ncovercash) [inline src/main/resources/swagger.api/schemas/FqmMigrateResponse.json:29]: see above comment
```suggestion
  "warnings": {
    "description": "List of warnings issued during the upgrade",
    "type": "array",
    "items": {
      "type": "object",
      "properties": {
        "description": { "type": "string" },
        "type": { "type": "string" }
      }
    }
  }
```

**Review by ncovercash — APPROVED** [review]

File: src/main/resources/swagger.api/mod-fqm-manager.yaml [hunk 3]
```diff
@@ -79,6 +79,30 @@ paths:
...
+        - fqmVersion
+      requestBody:
+        description: 'Request for FQM version submitted successfully'
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/fqmMigrateRequest'
+      responses:
+        '201':
```
Comment (ncovercash) [inline src/main/resources/swagger.api/mod-fqm-manager.yaml:96]: Probably 200

**Review by ncovercash — APPROVED** [review]

File: src/main/resources/swagger.api/mod-fqm-manager.yaml (hunk 3 above)
Comment (kjain110) [reply src/main/resources/swagger.api/mod-fqm-manager.yaml]: Nice catch! Done

**Pull request merged**